PlayerMovement.cs

```diff
@@ -1,180 +1,166 @@
 
     void Awake() {
-        rb = GetComponent<Rigidbody>();
+        rigidBody = GetComponent<Rigidbody>();
     }
-    
+
     void Start() {
         playerScale =  transform.localScale;
+        LockCursor();
+    }
+
+    void LockCursor() {
         Cursor.lockState = CursorLockMode.Locked;
         Cursor.visible = false;
     }
 
-    
-    private void FixedUpdate() {
+    void FixedUpdate() {
         Movement();
     }
 
-    private void Update() {
+    void Update() {
         MyInput();
         Look();
     }
 
     /// <summary>
     /// Find user input. Should put this in its own class but im lazy
     /// </summary>
-    private void MyInput() {
-        x = Input.GetAxisRaw("Horizontal");
-        y = Input.GetAxisRaw("Vertical");
-        jumping = Input.GetButton("Jump");
-        crouching = Input.GetKey(KeyCode.LeftControl);
-      
+    void MyInput() {
+        inputDirection.x = Input.GetAxisRaw("Horizontal");
+        inputDirection.y = Input.GetAxisRaw("Vertical");
+        inputDirection.Normalize();
+
+        if (Input.GetButton("Jump"))
+        {
+            Jump();
+        }
+
         //Crouching
         if (Input.GetKeyDown(KeyCode.LeftControl))
             StartCrouch();
         if (Input.GetKeyUp(KeyCode.LeftControl))
             StopCrouch();
     }
 
-    private void StartCrouch() {
+    void StartCrouch() {
+        crouching = true;
+        // squash the player
         transform.localScale = crouchScale;
+        // move them up
         transform.position = new Vector3(transform.position.x, transform.position.y - 0.5f, transform.position.z);
-        if (rb.velocity.magnitude > 0.5f) {
+        // if player is moving at a fast enough speed
+        if (rigidBody.velocity.magnitude > 0.5f) {
+            // and on the ground
             if (grounded) {
-                rb.AddForce(orientation.transform.forward * slideForce);
+                // slide boost forward
+                rigidBody.AddForce(orientation.transform.forward * slideForce);
             }
         }
     }
 
-    private void StopCrouch() {
+    void StopCrouch() {
+        crouching = false;
+        // reset scale
         transform.localScale = playerScale;
+        // move down
         transform.position = new Vector3(transform.position.x, transform.position.y + 0.5f, transform.position.z);
     }
 
-    private void Movement() {
-        //Extra gravity
-        rb.AddForce(Vector3.down * Time.deltaTime * 10);
-        
-        //Find actual velocity relative to where player is looking
-        Vector2 mag = FindVelRelativeToLook();
-        float xMag = mag.x, yMag = mag.y;
+    void Movement() {
+        // Extra gravity
+        rigidBody.AddForce(Vector3.down * Time.deltaTime * 10);
 
-        //Counteract sliding and sloppy movement
-        CounterMovement(x, y, mag);
-        
-        //If holding jump && ready to jump, then jump
-        if (readyToJump && jumping) Jump();
+        ApplyFriction();
```

Comment:
Since we're storing user input and can more easily get the users velocity relative to the camera, this method no longer requires params.